pandas/core/util/hashing.py

```diff
@@ -276,3 +302,23 @@ def hash_array(vals, encoding='utf8', hash_key=None, categorize=True):
     vals *= np.uint64(0x94d049bb133111eb)
     vals ^= vals >> 31
     return vals
+
+
+def _hash_scalar(val, encoding='utf8', hash_key=None):
+    """
+    Hash scalar value
+
+    Returns
+    -------
+    1d uint64 numpy array of hash value, of length 1
+    """
+
+    if isnull(val):
+        # this is to be consistent with the _hash_categorical implementation
+        return np.array([np.iinfo(np.uint64).max], dtype='u8')
+
```

Comment:
yes, I can certainly do that check here as well. 

That is maybe better to keep the custom logic here, as the keyword added to `infer_dtype_from_scalar` would not be used anywhere else.